src/layers/tile/mod.rs

```diff
@@ -115,18 +136,27 @@ impl<'map> LayerTile<'map> {
     }
     /// Get a reference to the layer tile's referenced tileset.
     pub fn get_tileset(&self) -> &'map Tileset {
-        // SAFETY: `tileset_index` is guaranteed to be valid
-        &self.map.tilesets()[self.data.tileset_index]
+        match &self.data.tileset_location {
+            // SAFETY: `tileset_index` is guaranteed to be valid
+            TilesetLocation::Map(n) => &self.map.tilesets()[*n],
+            TilesetLocation::Template(t) => &t,
+        }
     }
 
     /// Get the layer tile's tileset index. Guaranteed to be a
     /// valid index of the map tileset container, but **isn't guaranteed to actually contain
     /// this tile**.
     ///
+    /// Will return None if the tile is not located in one of the map's tilesets (i.e. the
+    /// tileset is from a template).
+    ///
     /// Use [`LayerTile::get_tile`] if you want to obtain the [`Tile`] that this layer tile is
     /// referencing.
-    pub fn tileset_index(&self) -> usize {
-        self.data.tileset_index
+    pub fn tileset_index(&self) -> Option<usize> {
+        match self.data.tileset_location {
+            TilesetLocation::Map(n) => Some(n),
+            TilesetLocation::Template(_) => None,
+        }
     }
```

Comment:
Now that `tileset_index` has been replaced by `tileset_location`, it makes more sense for this function to be replaced by a `tileset_location` readonly property.